module/module.go

```diff
@@ -480,6 +484,11 @@ func (m *Module) addAPIFromRegistry(ctx context.Context, api resource.Subtype) e
 
 // AddModelFromRegistry adds a preregistered component or service model to the module's services.
 func (m *Module) AddModelFromRegistry(ctx context.Context, api resource.Subtype, model resource.Model) error {
+	err := validateRegistered(api, model)
```

Comment:
Moved this to the top of the function, so if it errors, the API hasn't been added yet.